core/modulesource.go

```diff
@@ -42,14 +43,19 @@ func (proto ModuleSourceKind) ToLiteral() *idproto.Literal {
 }
 
 type ModuleSource struct {
+	Query *Query
+
 	Kind ModuleSourceKind `field:"true" name:"kind" doc:"The kind of source (e.g. local, git, etc.)"`
 
 	AsLocalSource dagql.Nullable[*LocalModuleSource] `field:"true" doc:"If the source is of kind local, the local source representation of it."`
-	AsGitSource   dagql.Nullable[*GitModuleSource]   `field:"true" doc:"If the source is a of kind git, the git source representation of it."`
 
-	// The root directory that was loaded for this module source, with any re-rooting
-	// based on the discovered path to the actual module configuration file.
-	RootDirectory dagql.Instance[*Directory] `field:"true" doc:"The root directory of the module source that contains its configuration and source code (which may be in a subdirectory of this root)."`
+	AsGitSource dagql.Nullable[*GitModuleSource] `field:"true" doc:"If the source is a of kind git, the git source representation of it."`
+
+	// Settings that can be used to initialize or override the source's configuration
+	WithName          string
+	WithDependencies  []dagql.Instance[*ModuleDependency]
+	WithSDK           string
+	WithSourceSubpath string
```

Comment:
> I guess these are distinctly named to indicate they're overrides to apply to the original module?

Yeah we do want to track them separately from the original values for various reasons:
* knowing the original name is important because it should always be used for codegen, with the override name only impacting the schema of those depending on it
* knowing the original dependencies is important since we de-dupe deps by symbolic ref with a preference for the ones specified in `WithDependencies`
* etc.

I'm trying to think of a better name. `OverrideName`/etc. feels wrong too these also are used when there wasn't an original value in the first place. If I think of something I'll update but otherwise leave it for now. There's a comment right above these lines explaining them if nothing else 😅